[circle-mlir] Revise README with how-to build (#14843)
Samsung/ONE: circle-mlir/README.md

# circle-mlir

Circle MLIR dialect and tools

> ## Tools provided

Reviewer: ```suggestion
## Provided Tools
```

_onnx2circle_

> - conversion tool of ONNX to Circle model for `compiler`

Reviewer: ```suggestion
- a tool that converts ONNX models to Circle format, used by `compiler`.
```

## How to build

Use provided `Makefile.sample` or create your own `Makefile`;
```
ln -s Makefile.sample Makefile
```
- `Makefile` is in `.gitignore` to let developers use own Makefile.

### Prerequisite

Install necessary packages;
```
sudo apt-get install build-essential cmake git
sudo apt-get install autoconf automake libtool unzip wget
sudo apt-get install devscripts debmake debhelper lcov
sudo apt-get install python3 python3-pip python3-venv python3-dev python3-all dh-python
```

Install `one-compiler` package;
- download and install latest `ONE Release` from https://github.com/Samsung/ONE/releases
- actually we only need `circle-interperter` tool for validation, but not ready yet

> Or use local build/install version of `ONE/compiler` and set environment variable;

Reviewer: ```suggestion
Alternatively, if you have a locally built version of `ONE/compiler`, set the environment variable as follows;
```

```
export ONE_COMPILER_ROOT=/home/user/one/build/install
```

### Prepare overlay

This will prepare virtual-env with necessary python packages installed;
```
make overlay
source infra/overlay/venv/bin/activate
```

NOTE `overlay` build is needed only once.

### Debug build

Build externals;
```
make prep
```

> NOTE `llvm-project` is built as `Debug` which may require 32G or more RAM.

Reviewer: ```suggestion
NOTE `llvm-project` builds in `Debug` mode, which may require 32GB or more RAM.
```

> - if build fails for some reason, please change to `-DCMAKE_BUILD_TYPE=Release`
>   `prep` target in `Makefile.sample` file.

Reviewer: ```suggestion
- if the build fails, try changing the build type to `-DCMAKE_BUILD_TYPE=Release`
  in the `prep` target of `Makefile.sample` file.
```
